src/Parser/Core/Modules/Items/EonarsCompassion.js

```diff
@@ -0,0 +1,84 @@
+import SPELLS from 'common/SPELLS';
+import ITEMS from 'common/ITEMS';
+import React from 'react';
+import Analyzer from 'Parser/Core/Analyzer';
+import Combatants from 'Parser/Core/Modules/Combatants';
+import { formatNumber, formatPercentage } from 'common/format';
+
+const PANTHEON_MAX_SHIELD_PER_PROC = 4;
+
+class EonarsCompassion extends Analyzer {
+  static dependencies = {
+      combatants: Combatants,
+  };
+  trinketHealing = 0;
+  trinketProc = 0;
+  pantheonShieldHealing = 0;
+  pantheonProc = 0;
+  pantheonShieldCast = 0;
+
+  triggerBuffs = [
+    SPELLS.EONARS_COMPASSION_PANTHEONBUFF_RDROOD.id,
+    SPELLS.EONARS_COMPASSION_PANTHEONBUFF_RMONK.id,
+    SPELLS.EONARS_COMPASSION_PANTHEONBUFF_HPALADIN.id,
+    SPELLS.EONARS_COMPASSION_PANTHEONBUFF_DPRIEST.id,
+    SPELLS.EONARS_COMPASSION_PANTHEONBUFF_HPRIEST.id,
+    SPELLS.EONARS_COMPASSION_PANTHEONBUFF_RSHAMAN.id,
+  ]
+
+  on_initialized() {
+    this.active = this.combatants.selected.hasTrinket(ITEMS.EONARS_COMPASSION.id);
+  }
+
+  on_byPlayer_heal(event) {
+    const spellId = event.ability.guid;
+    if (spellId === SPELLS.EONARS_COMPASSION_HEAL.id) {
+      this.trinketHealing += (event.amount || 0) + (event.absorbed || 0);
+    }
+  }
+
+  on_byPlayer_absorbed(event) {
+    const spellId = event.ability.guid;
+    if (spellId === SPELLS.EONARS_COMPASSION_PANTHEONSHIELD.id) {
+      this.pantheonShieldHealing += (event.amount || 0) + (event.absorbed || 0);
```

Comment:
I don't think absorbed events ever have an `absorbed` field, though in this case you've wrapped it with an `|| 0` so it doesn't matter much